### PR TITLE
Check cluster ID during service export

### PR DESCRIPTION
**Pull request by bansal19**
*Issue #, if available:* #254

*Description of changes:*
When exporting a service from Cluster A, we call DiscoverInstances on an entire clusterset in order to determine if there are any changes for the service we wish to export. If there are any changes, all endpoints in that service will be wiped by the endpoints being exported by Cluster A for this service. In order to mitigate this, we must check for the current cluster's endpoints **only** when calling DiscoverInstances to Cloud Map. 

To make sure we obtain the correct endpoints, we will need to invalidate the cache. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cloudmap/client.go [hunk 1]
```diff
@@ -112,20 +112,13 @@ func (sdc *serviceDiscoveryClient) CreateService(ctx context.Context, nsName str
 
 func (sdc *serviceDiscoveryClient) GetService(ctx context.Context, nsName string, svcName string) (svc *model.Service, err error) {
 	sdc.log.Info("fetching a service", "namespace", nsName, "name", svcName)
-	if endpts, found := sdc.cache.GetEndpoints(nsName, svcName); found {
```
Comment (runakash) [inline pkg/cloudmap/client.go:115]: Why is this cache call removed?

File: pkg/cloudmap/client.go [hunk 2]
```diff
@@ -208,6 +201,17 @@ func (sdc *serviceDiscoveryClient) getEndpoints(ctx context.Context, nsName stri
...
+	if err != nil {
+		sdc.log.Error(err, "failed to retrieve clusterSetId")
+		return nil, err
+	}
+	sdc.cache.CacheEndpoints(nsName, svcName, endpts)
+
+	return endpts, nil
+}
+
+func (sdc *serviceDiscoveryClient) getEndpointsInThisCluster(ctx context.Context, nsName string, svcName string, inThisCluster bool) (endpts []*model.Endpoint, err error) {
```
Comment (runakash) [inline pkg/cloudmap/client.go:214]: Lets be more explicit in the method declaration, keep the existing method `getEndpoints` and add `getCurrentClusterEndpoints` for readability.

File: pkg/cloudmap/client.go [hunk 3]
```diff
@@ -230,7 +237,6 @@ func (sdc *serviceDiscoveryClient) getEndpoints(ctx context.Context, nsName stri
 		}
 		endpts = append(endpts, endpt)
 	}
-	sdc.cache.CacheEndpoints(nsName, svcName, endpts)
```
Comment (runakash) [inline pkg/cloudmap/client.go:233]: Why is this removed?

File: pkg/cloudmap/client.go (hunk 1 above)
Comment (bansal19) [reply pkg/cloudmap/client.go:115]: Discussed offline, cache call was removed because it still contained endpoints which were not part of current cluster but part of cluster set. 
Solution: Since we are not dealing with huge services (max size 1000), we will still return cached endpoints filtered on `endpts.CLUSTER_ID` attribute

File: pkg/cloudmap/client_test.go [hunk 4]
```diff
@@ -283,24 +280,10 @@ func TestServiceDiscoveryClient_GetService_HappyCase(t *testing.T) {
...
-	assert.Nil(t, err)
-	assert.Equal(t, test.GetTestService(), svc)
-}
-
-func TestServiceDiscoveryClient_GetService_CachedValues(t *testing.T) {
-	tc := getTestSdClient(t)
-	defer tc.close()
-
-	tc.mockCache.EXPECT().GetEndpoints(test.HttpNsName, test.SvcName).
-		Return([]*model.Endpoint{test.GetTestEndpoint1(), test.GetTestEndpoint2()}, true)
```
Comment (bansal19) [inline pkg/cloudmap/client_test.go:303]: Bringing these tests back as well since we are keeping cache.

**Review by runakash — APPROVED** [review]

**Review by CurtisThe — APPROVED** [review]

**Review by astaticvoid — APPROVED** [review]
Thanks for the fix and improvements :)

**Review by jaywasd — APPROVED** [review]
